types/views/views.go

```diff
@@ -14,6 +14,78 @@ import (
 	"tailscale.com/net/tsaddr"
 )
 
+func unmarshalJSON[T any](b []byte, x *[]T) error {
+	if *x != nil {
+		return errors.New("already initialized")
+	}
+	if len(b) == 0 {
+		return nil
+	}
+	return json.Unmarshal(b, x)
+}
+
+// View represents the corresponding View of a Viewable. The concrete types are
```

Comment:
clarify this is always a struct and that slices use ViewSlice?

StructView? ViewStruct?

Reply:
actually it's a pointer to struct!